### PR TITLE
Fix mixed CP+IFL cpu_total and cpu_used from smcli 

### PR DESCRIPTION
**Pull request by xkwangcn**
Fix cpu total and cpu used number error in mixed CP+IFL env.

Get result from smcli to override the result from "cat /proc/sysinfo".

### CODE REVIEW

**Comment by xkwangcn** [comment]
1. output from smcli System_Processor_Query
```
[root@ssi013 smtLayer]# smcli System_Processor_Query
Partition mode: Z/VM

ADDRESS STATUS TYPE CORE_ID
0000 MASTER-PROCESSOR CP 0000
0002 ALTERNATE IFL 0001
0003 ALTERNATE IFL 0001
0004 ALTERNATE IFL 0002
0005 ALTERNATE IFL 0002
0006 ALTERNATE IFL 0003
0007 ALTERNATE IFL 0003
0008 ALTERNATE IFL 0004
0009 ALTERNATE IFL 0004
000A PARKED IFL 0005
000B PARKED IFL 0005
000C PARKED IFL 0006
000D PARKED IFL 0006
000E PARKED IFL 0007
000F PARKED IFL 0007
0010 PARKED IFL 0008
0011 PARKED IFL 0008
0012 PARKED IFL 0009
0013 PARKED IFL 0009
0014 PARKED IFL 000A
0015 PARKED IFL 000A
0016 PARKED IFL 000B
0017 PARKED IFL 000B
0018 PARKED IFL 000C
0019 PARKED IFL 000C
```
2. run the code in getHost.py
```
[root@ssi013 smtLayer]# python smtCmd.py GetHost general
ZCC USERID: SSI1009D
z/VM Host: BOEIAAS9
Architecture: s390x
CEC Vendor: IBM
CEC Model: 2964
Hypervisor OS: z/VM 7.2.0
Hypervisor Name: BOEIAAS9
LPAR CPU Total: 25
LPAR CPU Used: 25
LPAR Memory Total: 512G
LPAR Memory Offline: 0
LPAR Memory Used: 10.7G
IPL Time: IPL at 05/04/22 17:47:41 CES
```
3. test cases result:
```
[root@ssi013 smtLayer]# python -m unittest tests/unit/test_getHost.py -v
test_getCPUCount_noTypebutOOO (tests.unit.test_getHost.SMTGetHostTestCase) ... return value4: 0 0
ok
test_getCPUCount_no_CPU (tests.unit.test_getHost.SMTGetHostTestCase) ... return value3: 0 0
ok
test_getCPUCount_with_STANDBY (tests.unit.test_getHost.SMTGetHostTestCase) ... return value2: 25 22
ok
test_getCPUCount_with_overallRC_error (tests.unit.test_getHost.SMTGetHostTestCase) ... return value5: 0 0
ok
test_getCPUCount_without_STANDBY (tests.unit.test_getHost.SMTGetHostTestCase) ... return value1: 25 25
ok
test_getDiskPoolSpace3390 (tests.unit.test_getHost.SMTGetHostTestCase) ... ok
test_getDiskPoolSpace3390Unknown (tests.unit.test_getHost.SMTGetHostTestCase) ... ok
test_getDiskPoolSpace3390UnknownWithFlag (tests.unit.test_getHost.SMTGetHostTestCase) ... ok
test_getDiskPoolSpace9336 (tests.unit.test_getHost.SMTGetHostTestCase) ... ok
test_getDiskPoolSpace9336Unknown (tests.unit.test_getHost.SMTGetHostTestCase) ... ok
test_getDiskPoolSpace9336UnknownWithFlag (tests.unit.test_getHost.SMTGetHostTestCase) ... ok

----------------------------------------------------------------------
Ran 11 tests in 0.003s

OK
```
about the status(PARKED...) document: https://www.ibm.com/docs/en/zvm/7.2?topic=commands-query-processors, it decided the result for cpu_used.

File: smtLayer/getHost.py [hunk 1]
```diff
@@ -366,6 +366,51 @@ def getFcpDevices(rh):
...
+                    type_row = line_value.split(' ')[1]
+                    if type_row == 'TYPE':
+                        flag=1
+                    if flag==1:
+                        status_row = line_value.split(' ')[0]
+                        if(status_row.find('MASTER')!=-1 or status_row == 'ALTERNATE' or status_row =='PARKED'):
+                            cpu_use = cpu_use+1
+                        if(type_row=='CP' or type_row=='IFL'):
+                            cpu_total = cpu_total+1
+    else:
```
Comment (bjhuangr) [inline smtLayer/getHost.py:409]: if the zvm env no System_Processor_Query API, then error result will be written into request handler, does the `getHost general` will return error in this case ?

**Review by bjhuangr — COMMENTED** [review]
And you need to `git commit -s` to sign your commit

File: smtLayer/getHost.py (hunk 1 above)
Comment (xkwangcn) [reply smtLayer/getHost.py]: removed code for error result.

**Comment by xkwangcn** [comment]
Test on machine without System_Processor_Query API:
```
[root@dycomput smtLayer]# smcli System_Processor_Query
Failed
  Return Code: 900
  Reason Code: 12
  Description: ULGSMC5900E Specified function does not exist
  API issued : System_Processor_Query
[root@dycomput smtLayer]# python smtCmd.py GetHost general
ZCC USERID: DYCOMPUT
z/VM Host: BOEIAAS1
Architecture: s390x
CEC Vendor: IBM
CEC Model: 2964
Hypervisor OS: z/VM 7.1.0
Hypervisor Name: BOEIAAS1
LPAR CPU Total: 12
LPAR CPU Used: 12
LPAR Memory Total: 512G
LPAR Memory Offline: 0
LPAR Memory Used: 140.2G
IPL Time: IPL at 02/25/22 15:51:51 CES
[root@dycomput smtLayer]# python -m unittest tests/unit/test_getHost.py -v
test_getCPUCount_noTypebutOOO (tests.unit.test_getHost.SMTGetHostTestCase) ... return value4: 0 0
ok
test_getCPUCount_no_CPU (tests.unit.test_getHost.SMTGetHostTestCase) ... return value3: 0 0
ok
test_getCPUCount_with_STANDBY (tests.unit.test_getHost.SMTGetHostTestCase) ... return value2: 25 22
ok
test_getCPUCount_with_overallRC_error (tests.unit.test_getHost.SMTGetHostTestCase) ... return value5: 0 0
ok
test_getCPUCount_without_STANDBY (tests.unit.test_getHost.SMTGetHostTestCase) ... return value1: 25 25
ok
test_getDiskPoolSpace3390 (tests.unit.test_getHost.SMTGetHostTestCase) ... ok
test_getDiskPoolSpace3390Unknown (tests.unit.test_getHost.SMTGetHostTestCase) ... ok
test_getDiskPoolSpace3390UnknownWithFlag (tests.unit.test_getHost.SMTGetHostTestCase) ... ok
test_getDiskPoolSpace9336 (tests.unit.test_getHost.SMTGetHostTestCase) ... ok
test_getDiskPoolSpace9336Unknown (tests.unit.test_getHost.SMTGetHostTestCase) ... ok
test_getDiskPoolSpace9336UnknownWithFlag (tests.unit.test_getHost.SMTGetHostTestCase) ... ok

----------------------------------------------------------------------
Ran 11 tests in 0.003s

OK
```

**Review by bjhuangr — APPROVED** [review]

File: smtLayer/getHost.py [hunk 2]
```diff
@@ -367,6 +367,52 @@ def getFcpDevices(rh):
     return rh.results['overallRC']
 
 
+def getCPUCount(rh):
+    """
+    Obtain general information about the host.
```
Comment (SeanHQF) [inline smtLayer/getHost.py:372]: The description is not so precise, should be like 'get processor ...', right? anyway, not a big deal.

**Pull request merged**